src/main/java/io/jenkins/plugins/eggplant/EggplantRunnerBuilder.java

```diff
@@ -400,4 +472,140 @@ public static boolean isValidPath(String path, String extension) {
    
     }
 
+    public static abstract class TestConfig implements ExtensionPoint, Describable<TestConfig> {
+
+        public Descriptor<TestConfig> getDescriptor() {
+            return Jenkins.get().getDescriptor(getClass());
+        }
+    }
+
+    public static class TestConfigDescriptor extends Descriptor<TestConfig> {
+    } 
+
+    public static class TestConfigId extends TestConfig {
+        private final String id;
+
+        @DataBoundConstructor 
+        public TestConfigId(String id) {
+            this.id=id;
+        }
+
+        public String getId() {
+            return id;
+        }
+
+        @Extension
+        public static final class DescriptorImpl extends TestConfigDescriptor {
+
+            @Override
+            public String getDisplayName() {
+                return "By Id";
+            }
+
+            public FormValidation doCheckId(@QueryParameter String value) throws IOException {
+                if(value.isEmpty()) {
+                    return FormValidation.error("Test Config Id cannot be empty.");
+                }
+                else if(!isValidUuid(value)){
+                    return FormValidation.error("Invalid test configuration id.");
+                }
+                return FormValidation.ok();
+            }
+
+            private Boolean isValidUuid(String value){
+                Pattern p = Pattern.compile("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$");
+                Boolean isMatch=p.matcher(value).matches();
+                if(isMatch)
+                    return true;
+                else
+                    return false;
+            }
+
+        }
+    }
+
+
+    public static class ScriptBased extends TestConfig {
+        private final String name;
+        private final String suite;
+        @DataBoundConstructor public ScriptBased(String name, String suite) {
+            this.name=name;
+            this.suite=suite;
+        }
+        public String getName() {
+            return name;
+        }
+
+        public String getSuite() {
+            return suite;
+        }
+
+        @Extension
+        public static final class DescriptorImpl extends TestConfigDescriptor {
```

Comment:
```suggestion
        public static final class DescriptorImpl extends Descriptor<TestConfig> {
```

I will suggest to direct use the Descriptor<TestConfig> , straight forward and clear cut. No need to maintain another class.

Reply:
Sure, we can do that. Changes are updated.